arviz/tests/base_tests/test_plots_matplotlib.py

```diff
@@ -107,6 +107,25 @@ def data_list():
     return list(range(11, 31))
 
 
+@pytest.mark.parametrize(
+    "kwargs",
+    [
+        {"point_estimate": "mean"},
+        {"hdi_prob": 0.94},
+        {"outline": True},
+        {"colors": ["g", "b", "r", "y"]},
+        {"colors": "k"},
+        {"hdi_markers": ["v"]},
+        {"shade": 1},
+    ],
+)
+@pytest.mark.parametrize("filter_vars", [None, "like", "regex"])
```

Comment:
The test should check that filter vars is behaving as expected, especially when there are multiple inferencedata objects as inputs, we already have tests for the other cases. Now thinking about these tests I realized in order to keep tests short and to the point, it is best to test only the behaviour of `_var_names`, no need to run it _through_ plot_density. 

You can use the pattern in https://github.com/arviz-devs/arviz/blob/main/arviz/tests/base_tests/test_utils.py#L69 (with less cases, 2 per value of filter_var is probably enough) but using an iterable of 2 inferencedata as the 2nd (data) input. These two inferencedata inputs should have some variable names in common but also some that differ (there is also no need for as many variables in the first example, 2 in common and 1 different for each should be good)

I'd also recommend using `plot_density` after installing your local version on an interactive interpreter or notebook before writing the tests.